synthesizer/src/vm/finalize.rs

```diff
@@ -791,17 +792,66 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
                     //     }
                     // }
 
-                    // Initialize the stakers.
-                    let mut stakers = IndexMap::with_capacity(committee.members().len());
-                    // Iterate over the committee members.
-                    for (validator, (microcredits, _)) in committee.members() {
-                        // Insert the validator into the stakers.
-                        stakers.insert(*validator, (*validator, *microcredits));
+                    // Ensure that all members of the committee are open.
+                    ensure!(
+                        committee.members().iter().all(|(_, (_, is_open))| *is_open),
+                        "Ratify::Genesis(..) all members of the committee must be open"
+                    );
+
+                    // Calculate the stake per validator using `bonded_balances`.
+                    let mut stake_per_validator = IndexMap::with_capacity(committee.members().len());
+                    for (address, (validator_address, amount)) in bonded_balances {
+                        // Check that the amount meets the minimum requirement, depending on whether the address is a validator.
+                        if *address == *validator_address {
+                            ensure!(
+                                *amount >= MIN_VALIDATOR_STAKE,
+                                "Ratify::Genesis(..) the validator {address} must stake at least {MIN_VALIDATOR_STAKE}",
+                            );
+                        } else {
+                            ensure!(
+                                *amount >= MIN_DELEGATOR_STAKE,
+                                "Ratify::Genesis(..) the delegator {address} must stake at least {MIN_DELEGATOR_STAKE}",
+                            );
+                            // If the address is a delegator, check that the corresponding validator is open.
+                            ensure!(
+                                committee.is_committee_member_open(*validator_address),
+                                "Ratify::Genesis(..) the delegator {address} is delegating to a closed validator {validator_address}",
+                            );
+                        }
+                        // Ensure that each staker has an entry in `public_balances`.
+                        ensure!(
+                            public_balances.contains_key(address),
+                            "Ratify::Genesis(..) the staker {address} is missing from the public balances",
+                        );
```

Comment:
Why is this check necessary? Unless you decrement from `public_balances`, remove this check, it is extraneous.

Reply:
This check was introduced to be consistent with `credits.aleo`.
In the finalize block of `bond_public`, we [check](https://github.com/AleoHQ/snarkVM/blob/2c3b775f9c869e8ba03d10e91b94d8bf8297d388/synthesizer/program/src/resources/credits.aleo#L163) that a corresponding entry in the `account` mapping exists for both validators and delegators.